app/domain/operations/products/import_rating_area.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+module Operations
+  module Products
+    # This class is to load county zip combinations.
+    class ImportRatingArea
+      include Dry::Monads[:result, :do]
+
+      def call(params)
+        values  = yield validate(params)
+        data    = yield load_data(values[:file])
+        _load = yield import_records(data, values[:year], values[:import_timestamp])
+        Success()
+      end
+
+      private
+
+      def validate(params)
+        return Failure('Missing File') if params[:file].blank?
+        return Failure('Missing Year') if params[:year].blank?
+        return Failure('Missing Import TimeStamp') if params[:import_timestamp].blank?
+        Success(params)
+      end
+
+      def load_data(file)
+        geographic_rating_area_model = EnrollRegistry[:enroll_app].setting(:geographic_rating_area_model).item
+        result = Roo::Spreadsheet.open(file)
+        sheet = result.sheet(0)
+        result = Hash.new {|results, k| results[k] = []}
+
+        case geographic_rating_area_model
+        when 'single'
+          result = {}
+        when 'county'
+          (2..sheet.last_row).each do |i|
+            result[sheet.cell(i, 4)] << { 'county_name' => sheet.cell(i, 2) }
+          end
+        when 'zipcode'
+          (2..sheet.last_row).each do |i|
+            result[sheet.cell(i, 4)] << { 'zip' => sheet.cell(i, 1) }
+          end
+        else
+          (2..sheet.last_row).each do |i|
+            result[sheet.cell(i, 4)] << {
+              'county_name' => sheet.cell(i, 2),
+              'zip' => sheet.cell(i, 1)
+            }
+          end
+        end
+        Success(result)
+      end
+
+      def import_records(data, year, import_timestamp)
+        state_abbreviation = Settings.aca.state_abbreviation
+        geographic_rating_area_model = EnrollRegistry[:enroll_app].setting(:geographic_rating_area_model).item
```

Comment:
In the case of a single rating area you still need to create a single, state wide rating area.

Reply:
addressed.